x-pack/plugins/security_solution/common/types/header_actions/index.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  EuiDataGridCellValueElementProps,
+  EuiDataGridColumn,
+  EuiDataGridColumnCellActionProps,
+  EuiDataGridControlColumn,
+} from '@elastic/eui';
+import type { IFieldSubType } from '@kbn/es-query';
+import type { FieldBrowserOptions } from '@kbn/triggers-actions-ui-plugin/public';
+import type { ComponentType, JSXElementConstructor, ReactNode } from 'react';
+import type { OnRowSelected, SetEventsDeleted, SetEventsLoading } from '..';
+import type { Ecs } from '../../ecs';
+import type { BrowserFields, TimelineNonEcsData } from '../../search_strategy';
+import type { SortColumnTable } from '../data_table';
+
+export type ColumnHeaderType = 'not-filtered' | 'text-filter';
+
+/** Uniquely identifies a column */
+export type ColumnId = string;
+
+/**
+ * A `DataTableCellAction` function accepts `data`, where each row of data is
+ * represented as a `TimelineNonEcsData[]`. For example, `data[0]` would
+ * contain a `TimelineNonEcsData[]` with the first row of data.
+ *
+ * A `DataTableCellAction` returns a function that has access to all the
+ * `EuiDataGridColumnCellActionProps`, _plus_ access to `data`,
+ *  which enables code like the following example to be written:
+ *
+ * Example:
+ * ```
+ * ({ data }: { data: TimelineNonEcsData[][] }) => ({ rowIndex, columnId, Component }) => {
+ *   const value = getMappedNonEcsValue({
+ *     data: data[rowIndex], // access a specific row's values
+ *     fieldName: columnId,
+ *   });
+ *
+ *   return (
+ *     <Component onClick={() => alert(`row ${rowIndex} col ${columnId} has value ${value}`)} iconType="heart">
+ *       {'Love it'}
+ *      </Component>
+ *   );
+ * };
+ * ```
+ */
+export type DataTableCellAction = ({
+  browserFields,
+  data,
+  ecsData,
+  header,
+  pageSize,
+  scopeId,
+  closeCellPopover,
+}: {
+  browserFields: BrowserFields;
+  /** each row of data is represented as one TimelineNonEcsData[] */
+  data: TimelineNonEcsData[][];
+  ecsData: Ecs[];
+  header?: ColumnHeaderOptions;
+  pageSize: number;
+  scopeId: string;
+  closeCellPopover?: () => void;
+}) => (props: EuiDataGridColumnCellActionProps) => ReactNode;
+
+/** The specification of a column header */
+export type ColumnHeaderOptions = Pick<
```

Comment:
Same for this type as well

Reply:
By the reason I've mentioned above - we need this type for hover actions and Observability still use it by some reason.